kos_sim/services.py

```diff
@@ -197,3 +218,42 @@ def GetQuaternion(  # noqa: N802
             context.set_code(grpc.StatusCode.INTERNAL)
             context.set_details(str(e))
             return imu_pb2.QuaternionResponse()
+
+    def GetEuler(  # noqa: N802
+        self, request: empty_pb2.Empty, context: grpc.ServicerContext
+    ) -> imu_pb2.EulerAnglesResponse:
+        """Implements GetEuler by converting orientation quaternion to Euler angles."""
+        try:
+            quat_data = self.simulator.get_sensor_data("orientation")
+            # Extract quaternion components
+            w, x, y, z = [float(q) for q in quat_data]
```

Comment:
wait, where does `w x y z` instead of `x y z w` come from? is this mujoco thing?

i think we use the other convention usually

https://github.com/kscalelabs/kos/blob/master/kos/proto/kos/imu.proto#L83-L89

annoying bug if done wrong

should add a comment here at least